src/funcspace_to_vecspace.cpp

```diff
@@ -0,0 +1,70 @@
+/*! \file fasp_to_fenics.cpp
+ *
+ * \brief Contains functions to transfer EigenMatrix and EigenVector to FASP
+ *  or to tranfer FASP to Fenics
+ */
+
+#include <iostream>
+#include <dolfin.h>
+#include "funcspace_to_vecspace.h"
+
+using namespace std;
+using namespace dolfin;
+
+/**
+ * \fnvoid add_matrix(int l, dolfin::FunctionSpace *V, dolfin::FunctionSpace *V_l, dolfin::EigenMatrix *A, dolfin::EigenMatrix *A_l)
+ *
+ * \brief add A_l to A for l=0,1,2 is the lth component
+ *
+ */
+void add_matrix(int l, dolfin::FunctionSpace *V, dolfin::FunctionSpace *V_l, dolfin::EigenMatrix *A, dolfin::EigenMatrix *A_l)
+{
+  // dimensions of matrices
+  int nrows_l = A_l->size(0);
+  int ncols_l = A_l->size(1);
+  int nrows = A->size(0);
+  int ncols = A->size(1);
+
+  // point to JA/IA/Values array
+  int* JA_l=(int*) std::get<1>(A_l->data());
+  int *IA_l=(int*) std::get<0>(A_l->data());
```

Comment:
Sorry to be a pain in the ass, but it would be good to have consistent syntax:
``int* IA_l = (int*) std::get<0> (A_l->data());``
Same thing for declaring ``add_matrix`` and line 33